### ISSUE TITLE
Tracking scope change

### ISSUE THREAD

**Issue by bradical** [ZenHubIO/support#791]
Is there any way to track scope change during a milestone? 

Specifically, i'm looking for lists of issues added/removed/change and the delta for any points added/removed.

If not, I would 👍 any feature that did this.

Thanks!

**Comment by azenMatt** [comment]
Hi @bradical - not yet! 

But I agree this would be a very useful report - and could be wrapped into a bunch of other 'release information' for a milestone dashboard.

**Comment by stevenmccord** [comment]
@azenMatt would this feature also include release date changes?  I can see something like hey we were all striving for 2/1, but now we need to re-scope that to 2/15.  Not sure if you guys are already thinking about that or not.  Thanks!

**Comment by azenMatt** [comment]
Hi, we are looking to implement something like the following to track the change of issue scope in a release, any feedback on if it would be helpful? 

This is based on making sure we can track: 


- As a user, I want to understand when **issues have been removed from scope** during the course of a sprint so I can help the team get better at planning sprints 
- As a user, I want to understand which **issues have been added to scope** during the course of a sprint so I can help the team get better at planning sprints 
- As a user, I want  to understand when **issues are left incomplete at the end of a milestone**, even if they are moved to a separate milestone so I can see where the team over-commit. (note - this is similar to ‘remove from scope’ but with an added emphasis on ‘roll over’ to the next sprint)
- As a user, I want to understand when **issues are re-estimated** during a milestone so I can help the team become better at estimation up front


![image](https://user-images.githubusercontent.com/7584089/35892659-3cf86af2-0b5f-11e8-817e-31690a7fc915.png)

**Comment by stevenmccord** [comment]
@azenMatt thanks for the read out.  You mention milestone in the mocks, are you guys applying this to releases as well?  We utilize releases as we have tickets in a bunch of different repos and that is how we are tracking that information.  Thanks!

**Comment by bradical** [comment]
It looks like they may be doing a bit of this for Releases already. I
thought I saw information about issues added/removed from scope in a
release recently.

**Comment by azenMatt** [comment]
Hi @stevenmccord @bradical - the above designs are only for the burndown at the moment. 

For releases, they should already have much of this audit information in the issue list. For example, if I change the end date of a release:

![image](https://user-images.githubusercontent.com/7584089/35929624-c31ba22c-0be4-11e8-87e5-e367fbac7b84.png)


But... this is packed in with a lot of other information related to scope change. So perhaps needs to be given more room to itself?

**Comment by stevenmccord** [comment]
Yeah I think that is good that you can see it in the activity, but probably too packed with information.  Is that available in the zenhub api?  I am wondering if I could create a custom report that would give the original date vs the end date, something to that effect.

**Comment by azenMatt** [comment]
unfortunately the historical information isn't available on the API yet

**Comment by chadfurman** [comment]
this somewhat misses the mark for me.  Lets say we have the same 5 issues, A B C D E at the start of the sprint for a total of 33 points.

At the end of the sprint, C and D are not done.  Now I close the milestone.  A B C D E are all still in the same milestone report as being done/not done etc.  

I then move C and D into milestone 2.  The report for milestone 1 should still show A B C D E, as well as their open/closed status at the end of the milestone.  The milestone is closed, so the report should not change after that.

I would like a snapshot-in-time of a milestone at the time I closed it.  I would also like a way of persisting my burndown chart selections so that all my milestones treat the "passed review" pipeline as "closed" -- or atleast that they remember it when I refresh the page...

**Comment by bradical** [comment]
Same here. I’ve taken to creating literal snapshots of the screen at this
state prior to moving those issues to the next milestone.

**Comment by azenMatt** [comment]
Thanks everyone for the additional feedback here - we are taking another
pass at it to see what we can do to make sure this works well for all the
different use cases

Updates to follow :)

**Comment by chadfurman** [comment]
How did the iteration go on this design?  This is very similar to https://github.com/ZenHubIO/support/issues/1283#issuecomment-374603268 right?

**Comment by azenMatt** [comment]
This is similar to #1283 - right now, we are working with the GH team to see if there is a way to get access to all the data required from their API to generate a report like the one above. 

If not, or the time frame is too long. I think that doing something closer to #1283 in the interim could make sense

**Comment by chadfurman** [comment]
@azenMatt #1283 makes sense to me as well.  Especially if for the complex chart above you need to have GH team make API changes -- freezing burndown charts will provide much-needed velocity history.  

If the freeze is relatively low LOE, I think it is the most intuitive solution anyways.  Once I close a milestone, the burndown for that milestone should not change -- moving tickets out of the milestone after it is closed logically does not affect the sprint so it should not affect the burndown.

The chart above makes sense for scope changes during an active sprint, I agree.

**Comment by chadfurman** [comment]
Any follow-up on this?  April, may, june, july, now August -- there seems to be a bit of interest in this feature from the above discussion, but I can understand why it might not be considered a priority
